internal/credential/state.go

```diff
@@ -20,9 +20,23 @@ import (
 type CredentialType int
 
 const (
-	Unknown CredentialType = iota
-	Static
-	Dynamic
+	// StaticAWS denotes an Access Key Id that begins with "AKIA". These are
+	// long-term access keys, provided by AWS, for an IAM user or an AWS account
+	// root user.
+	// https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html#sec-access-keys-and-secret-access-keys
+	StaticAWS CredentialType = iota
+
+	// DynamicAWS denotes the presence of a RoleARN, or an Access Key Id that
+	// begins with "ASIA". The latter are temporary credentials access keys that
+	// are created using AWS STS operations.
+	DynamicAWS
+
+	// StaticOther denotes the presence of an Access Key Id that does not follow
+	// the AKIA/ASIA convention outlined by AWS.
+	StaticOther
```

Comment:
I am curious if this could ever cause problems if returned by `GetCredentialType` since there are a few checks for `StaticAWS` specifically, or is this just for plugins other than AWS?

Reply:
> or is this just for service providers other than AWS?

Yep, this is exactly what it is for - `StaticAWS` uses the hard-coded AWS conventions, and then everything else that doesn't follow that is `StaticOther`, for S3-Compatible... compatibility 😂

Reply:
Have we tested a S3 compatible service with these code changes?

Reply:
I have not - I think that's outside the scope of this work since this patch isn't really about enabling S3-compatible services. The `StaticOther` credential type is introduced here more as a way of keeping that path open for future S3-Compatibility work.

I might spin up a MinIO instance and give it a quick shot though 😄